lib/notifier.js

```diff
@@ -256,10 +256,22 @@ class Listener {
 
 let listener;
 function addListener(server, user, regex, event, callback) {
+    let config = server; // Using the new API, server should contain a config object.
+
+    // Convert from the old API if needed.
+    if (arguments.length === 5) {
+        config = {
+            serverUrl: server,
+            adminUser:  user,
+            filterRegex: regex
+            // SSL Configuration not supported in the old API
+        }
+    }
+
     if (!listener) {
-        listener = new Listener(this, server, user);
+        listener = new Listener(this, config);
     }
-    return listener.add(regex, event, callback);
+    return listener.add(config.filterRegex, event, callback);
```

Comment:
With the new API, wouldn't `user` and `regex` effectively be `event` and `callback`?

Reply:
Yes, good catch 👍